### PR TITLE
chore(client): update to webpack@5

### PR DESCRIPTION
**Pull request by marstamm**
closes https://github.com/camunda/camunda-modeler/issues/3060

### CODE REVIEW

File: client/karma.config.js [hunk 1]
```diff
@@ -105,6 +106,9 @@ module.exports = function(karma) {
         ]
       },
       plugins: [
+        new webpack.ProvidePlugin({
+          process: 'process/browser',
```
Comment (nikku) [inline client/karma.config.js:110]: What do we need this for and can't we get rid of it with `DefinePlugin`?

File: client/webpack.config.js [hunk 2]
```diff
@@ -37,23 +37,33 @@ const { LicenseWebpackPlugin } = require('license-webpack-plugin');
...
   resolve: {
-    mainFields: DEV ? [ 'browser', 'dev:module', 'module', 'main' ] : undefined,
+    mainFields: DEV ? [ 'browser', 'dev:module', 'module', 'main' ] : [ 'browser', 'module', 'main' ],
     modules: [
       'node_modules',
       resourcePath
-    ]
+    ],
+    fallback: {
+      util: require.resolve('util/')
```
Comment (nikku) [inline client/webpack.config.js:65]: What do we need this for?

File: client/webpack.config.js [hunk 3]
```diff
@@ -127,7 +134,7 @@ module.exports = {
   devServer: {
     writeToDisk: true
   },
-  devtool: DEV ? 'cheap-module-eval-source-map' : 'source-map'
+  devtool: DEV ? 'cheap-module-source-map' : 'source-map'
```
Comment (nikku) [inline client/webpack.config.js:137]: Why? :)

Comment (marstamm) [reply client/webpack.config.js:137]: `cheap-module-eval-source-map` is not a valid devtool in webpack@5: https://webpack.js.org/configuration/devtool/

File: client/karma.config.js (hunk 1 above)
Comment (marstamm) [reply client/karma.config.js:110]: Webpack@5 no longer provides nodeJS polyfills for browsers by default. In the tests, we use [`process.nextTick()` ](https://github.com/camunda/camunda-modeler/blob/58123a652018db7d73ce8274d98713eece3f6020/client/src/app/__tests__/AppParentSpec.js#L466), which now needs to be provided by this plugin.

The Define plugin is still needed so we can supply env variables

File: client/webpack.config.js (hunk 2 above)
Comment (marstamm) [reply client/webpack.config.js:65]: Similar to https://github.com/camunda/camunda-modeler/pull/3062#discussion_r940995410, `util` is used in the client (e.g. [util/inherits](https://github.com/camunda/camunda-modeler/blob/736fe6cef9d6857b0beef4ee0c0482cba85b298f/client/src/app/tabs/dmn/modeler/features/decision-table-keyboard/DecisionTableKeyboard.js#L11)) and needs to be provided by us in Webpack@5

Comment (nikku) [reply client/webpack.config.js:65]: We should rather change to [`inherits-browser`](https://github.com/nikku/inherits-browser) which gets rid of the util import _or_ configure webpack to use the `browser` field over the `main` and `module` fields.

File: client/karma.config.js (hunk 1 above)
Comment (nikku) [reply client/karma.config.js:110]: Uhm... let's get rid of `process.nextTick?`

**Comment by marstamm** [comment]
I applied the suggested changes and removed the `process.nextTick` usage and switched to `inherits-browser`.

I also removed a broken check for linux that was always false and would break the test if fixed (cf. [04909ec](https://github.com/camunda/camunda-modeler/pull/3062/commits/04909ec73bf339195aa41f18365e7b79de89c197), [broken CI run](https://github.com/camunda/camunda-modeler/actions/runs/2859447636))

**Review by nikku — APPROVED** [review]
Awesome stuff. Great showing of incremental improvement on multiple levels!

**Pull request merged**